module/zfs/dmu_send.c

```diff
@@ -2448,7 +2456,9 @@ receive_object(struct receive_writer_arg *rwa, struct drr_object *drro,
 	}
 
 	if (rwa->raw) {
-		if (drro->drr_raw_bonuslen < drro->drr_bonuslen ||
```

Comment:
I know this is explained elsewhere, but this is subtle enough that I think it's worth a brief comment stating this is expected to be true because for a raw receive a `DRR_OBJECT_RANGE` record must have been handled which describes the encryption parameters.